### DISCUSSION TITLE
How to exclude folders

### DISCUSSION THREAD

**Discussion by ndeville** [getpelican/pelican#3111] [Q&A]
To speed up deployment time, I am now splitting hosting of my static files between Github Pages (HTML, CSS) and ImageKit (images & PDFs).

I can't manage to exclude my folders `images` and `pdfs` from being generated in the git-enabled output folder though.

I tried the following:

``` python
IGNORE_FILES = [
    "images/**/*.{jpg,jpeg,png,gif,svg}",
    "pdfs/*.pdf",
    ]

# IGNORE_FOLDERS = [
#     "images", 
#     "pdfs",
#     ]
```

What is the proper way to exclude folders? 🤔

**Comment by justinmayer** [comment]
Hi Nic. As noted in the [Pelican settings documentation](https://docs.getpelican.com/en/latest/settings.html), by default `STATIC_PATHS = ["images"]`, so everything within will be copied unless otherwise specified by adding folder paths to ignore to the `STATIC_EXCLUDES = []` setting.

As for the PDFs, I imagine replacing `"pdfs/*.pdf"` with just `"*.pdf"` should deliver the expected result.

Related pre-existing issue: #2979

**Reply by ndeville** [reply]
Thank you @justinmayer 

However, I could not make it work with `STATIC_EXCLUDES` either.

Tried with:

``` python
STATIC_EXCLUDES = [
    "images",
    "pdfs",
]
```
and
``` python
STATIC_EXCLUDES = [
    "images/**/*.{jpg,jpeg,png,gif,svg}",
    "pdfs/*.pdf",
]
```

Note that my `STATIC_PATHS` already excluded those folders:

``` python
STATIC_PATHS = [
                # 'images', 
                'extra', 
                # 'pdfs', 
                'pages',
                'csvs',
                ]
```

Not critical for me anymore though because I implemented the following workaround which gets triggered after generation:

``` python
from send2trash import send2trash

folder_images = '/xxxx/images'
if os.path.exists(folder_images) and os.path.isdir(folder_images):
    send2trash(folder_images)

folder_pdfs = '/xxxx/pdfs'
if os.path.exists(folder_pdfs) and os.path.isdir(folder_pdfs):
    send2trash(folder_pdfs)
```

**Comment by justinmayer** [comment]
As I mentioned in the pre-existing linked issue, I was able to exclude patterns but understand that didn't work for you for some reason. I don't have time at the moment to try to replicate the behavior you described, but I'm glad at least you found a suitable workaround.
